src/processFiles.js

```diff
@@ -0,0 +1,35 @@
+import { readFileSync, writeFileSync } from 'fs';
+import * as prettier from 'prettier';
+import { join } from 'path';
+
+export default (
+  directory,
+  files,
+  { check, config, onProcessFile, onExamineFile } = {}
```

Comment:
Names `onProcessFile`, `onExamineFile` are a little too similar for my liking.

How about:
- `onProcessFile` - file is being looked at
- `onWriteFile` - file was just written
- `onCheckFile` - file was just checked - always call with `(file, isFormatted)`

Reply:
This makes sense! Fixed.